internal/status/proxystatus_test.go

```diff
@@ -0,0 +1,313 @@
+			{
+				Condition: projectcontour.Condition{
+					Type:               string(ValidCondition),
+					Status:             projectcontour.ConditionFalse,
+					ObservedGeneration: testGeneration,
+					LastTransitionTime: testTransitionTime,
+					Reason:             "TLSErrorTLSConfigError",
+					Message:            "Syntax Error in TLS Config",
+				},
+				Errors: []projectcontour.SubCondition{
+					{
+						Type:    "TLSError",
+						Reason:  "TLSConfigError",
+						Message: "Syntax Error in TLS Config",
+					},
+				},
+			},
+		},
+		wantCurrentStatus: k8s.StatusInvalid,
+		wantDescription:   "TLSErrorTLSConfigError: Syntax Error in TLS Config",
+	}
+	run("invalid status, one error", inValidConditionError)
+
+	orphanedCondition := testcase{
+		testProxy: projectcontour.HTTPProxy{
+			ObjectMeta: v1.ObjectMeta{
+				Name:       "test",
+				Namespace:  "test",
+				Generation: testGeneration,
+			},
+		},
+		proxyUpdate: ProxyUpdate{
+			Fullname:       k8s.NamespacedNameFrom("test/test"),
+			Generation:     testGeneration,
+			TransitionTime: testTransitionTime,
+			Conditions: map[ConditionType]*projectcontour.DetailedCondition{
+				ValidCondition: {
+					Condition: projectcontour.Condition{
+						Type:    string(ValidCondition),
+						Status:  projectcontour.ConditionFalse,
+						Reason:  "orphaned",
+						Message: "this HTTPProxy is not part of a delegation chain from a root HTTPProxy",
+					},
+					Errors: []projectcontour.SubCondition{
+						{
+							Type:    "orphaned",
+							Reason:  "Orphaned",
+							Message: "this HTTPProxy is not part of a delegation chain from a root HTTPProxy",
+						},
+					},
+				},
+			},
+		},
+		wantConditions: []projectcontour.DetailedCondition{
+			{
+				Condition: projectcontour.Condition{
+					Type:               string(ValidCondition),
+					Status:             projectcontour.ConditionFalse,
+					ObservedGeneration: testGeneration,
+					LastTransitionTime: testTransitionTime,
+					Reason:             "orphaned",
+					Message:            "this HTTPProxy is not part of a delegation chain from a root HTTPProxy",
+				},
+				Errors: []projectcontour.SubCondition{
+					{
+						Type:    "orphaned",
+						Reason:  "Orphaned",
+						Message: "this HTTPProxy is not part of a delegation chain from a root HTTPProxy",
+					},
+				},
+			},
+		},
+		wantCurrentStatus: k8s.StatusOrphaned,
+		wantDescription:   "this HTTPProxy is not part of a delegation chain from a root HTTPProxy",
+	}
+
+	run("orphaned HTTPProxy", orphanedCondition)
+
+	updateExistingValidCond := testcase{
+		testProxy: projectcontour.HTTPProxy{
+			ObjectMeta: v1.ObjectMeta{
+				Name:       "test",
+				Namespace:  "test",
+				Generation: testGeneration,
+			},
+			Status: projectcontour.HTTPProxyStatus{
+				Conditions: []projectcontour.DetailedCondition{
+					{
+						Condition: projectcontour.Condition{
+							Type:   string(ValidCondition),
+							Status: projectcontour.ConditionTrue,
```

Comment:
`Reason` is a required field ... what should it be when the condition is true?

Reply:
"valid", I'll update.